src/pages/clients/common/components/Settings.tsx

```diff
@@ -41,7 +41,11 @@ export default function Settings() {
   const handleChange = (property: string, value: string | number | boolean) => {
     const $client = cloneDeep(client)!;
 
-    if (property !== 'currency_id' && value === '') {
+    if (
+      property !== 'currency_id' &&
+      $client.settings?.[property] !== undefined &&
+      value === ''
+    ) {
       delete $client.settings?.[property];
```

Comment:
@beganovich I just took a look at this logic. I did not delete the old `AdditionalInfo` component and there is this logic of deleting property also. Is there any specific reason why the prop should be completely deleted from the client.settings object? I mean it is just changing the value...

Reply:
@Civolilah unsetting the prop is very important.

When a prop is SET at the group or client level it will override the company level prop.

If the prop does not exist, then it signifies that the property has not been set at the client/group level.

why is this important? At settings level you can have bool / null values as valid values, so there is no way to determine which property is "active" at this level.

Reply:
@turbo124 Ah okay, thanks for the explanation! I got it right now!